server/test/services/statuses/StatusServiceTest.java

```diff
@@ -0,0 +1,675 @@
+
+    assertThat(result.isError()).isFalse();
+    ProgramDefinition definition = result.getResult();
+    assertThat(definition.localizedName().get(Locale.GERMAN)).isEqualTo("German Name");
+    assertThat(definition.localizedDescription().get(Locale.GERMAN))
+        .isEqualTo("German Description");
+    assertThat(definition.localizedSummaryImageDescription().isPresent()).isTrue();
+    assertThat(definition.localizedSummaryImageDescription().get().get(Locale.GERMAN))
+        .isEqualTo("German Image Description");
+
+    ErrorAnd<StatusDefinitions, CiviFormError> result2 =
+        service.updateLocalization(program.id, Locale.GERMAN, updateData);
+
+    assertThat(result2.isError()).isFalse();
+    ImmutableList<StatusDefinitions.Status> expectedStatuses =
+        ImmutableList.of(
+            StatusDefinitions.Status.builder()
+                .setStatusText(STATUS_WITH_EMAIL.statusText())
+                .setLocalizedStatusText(
+                    STATUS_WITH_EMAIL
+                        .localizedStatusText()
+                        .updateTranslation(Locale.GERMAN, "german-status-with-email"))
+                .setLocalizedEmailBodyText(
+                    Optional.of(
+                        STATUS_WITH_EMAIL
+                            .localizedEmailBodyText()
+                            .get()
+                            .updateTranslation(Locale.GERMAN, "german email body")))
+                .build(),
+            StatusDefinitions.Status.builder()
+                .setStatusText(STATUS_WITH_NO_EMAIL.statusText())
+                .setLocalizedStatusText(
+                    STATUS_WITH_NO_EMAIL
+                        .localizedStatusText()
+                        .updateTranslation(Locale.GERMAN, "german-status-with-no-email"))
+                .build());
+    assertThat(result2.getResult().getStatuses()).isEqualTo(expectedStatuses);
+    assertThat(
+            applicationStatusesRepo
+                .lookupActiveStatusDefinitions(definition.adminName())
+                .getStatuses())
+        .isEqualTo(expectedStatuses);
+  }
+
+  @Test
+  public void updateLocalizations_updatesExistingLocale() throws Exception {
+    ProgramModel program =
+        ProgramBuilder.newDraftProgram("English name", "English description")
+            .withLocalizedName(Locale.FRENCH, "existing French name")
+            .withLocalizedDescription(Locale.FRENCH, "existing French description")
+            .withLocalizedConfirmationMessage(Locale.FRENCH, "")
+            .setLocalizedSummaryImageDescription(
+                LocalizedStrings.of(
+                    Locale.US,
+                    "English image description",
+                    Locale.FRENCH,
+                    "existing French image description"))
+            .withStatusDefinitions(
+                new StatusDefinitions(ImmutableList.of(STATUS_WITH_EMAIL, STATUS_WITH_NO_EMAIL)))
+            .build();
```

Comment:
Throughout this file, instead of using the `ProgramBuilder` to save statuses, can we save them directly using the `ApplicationStatusesRepository`? Then we don't need to deal with creating programs.

Since they were copied from the `ProgramService`, these tests are testing a lot of the program behavior unnecessarily.

Reply:
I think the problem arises because of using the same LocalizationUpdate param for both program update and status update. But I hear your point. Program data shouldn't be needed here. I plan to do a followup PR to remove status from programBuilder and get rid of everywhere. This PR is doing many things already, complexity of removing it will be more,